shoop/front/middleware.py

```diff
@@ -77,6 +80,9 @@ def _set_shop(self, request):
 
     def _set_person(self, request):
         request.person = get_person_contact(request.user)
+        if not request.person.is_active:
+            messages.add_message(request, messages.INFO, _("Logged out since this account is inactive."))
+            logout(request)
```

Comment:
This should also set the request.person to AnonymousContact. I'll create a PR for that.